bin/clinvar_jsons/parse_trait_names.py

```diff
@@ -1,52 +1,72 @@
 import argparse
+import csv
 import json
 import gzip
 import sys
-from collections import defaultdict
+from collections import defaultdict, namedtuple
 
 
 def main():
     parser = ArgParser(sys.argv)
 
-    trait_names = defaultdict(int)
-
+    trait_dict = {}
     for clinvar_json in clinvar_jsons(parser.infile_path):
-        var_trait_names = get_trait_names(clinvar_json)
-        for name in var_trait_names:
-            trait_names[name] += 1
+        if not is_allowed_clin_sig(clinvar_json):
+            continue
+        trait_dict = get_traits_from_json(clinvar_json, trait_dict)
+
+    with open(parser.outfile_path, "w", newline="") as outfile:
+        writer = csv.writer(outfile, delimiter="\t")
+        for trait in trait_dict.values():
+            writer.writerow([trait.name, trait.xref_string, trait.count])
+
 
-    with gzip.open(parser.outfile_path, "wt") as outfile:
-        for name, count in trait_names.items():
-            outfile.write(name + "\t" + str(count) + "\n")
+def is_allowed_clin_sig(clinvar_json):
+    allowed_clin_sigs = ["pathogenic", "likely pathogenic", "protective", "association",
+                         "risk_factor", "affects", "drug response"]
+    if "description" in clinvar_json["clinvarSet"]["referenceClinVarAssertion"]["clinicalSignificance"]:
+        if clinvar_json["clinvarSet"]["referenceClinVarAssertion"]["clinicalSignificance"]["description"].lower() in allowed_clin_sigs:
+            return True
+    return False
 
 
-def get_trait_names(clinvar_json):
+def get_traits_from_json(clinvar_json, trait_dict):
+
     # This if-else block is due to the change in the format of the CellBase JSON that holds the
     # ClinVar data. Originally "clinvarSet" was the top level, but this level was removed and
     # referenceClinVarAssertion is now the top level.
     if "clinvarSet" in clinvar_json:
         trait_set = clinvar_json["clinvarSet"]["referenceClinVarAssertion"]["traitSet"]
     else:
         trait_set = clinvar_json["referenceClinVarAssertion"]["traitSet"]
-    trait_list = []
-    for trait in trait_set['trait']:
-        trait_list.append([])
-        for name in trait['name']:
+    for trait_doc in trait_set['trait']:
```

Comment:
This is to distinguish between the "trait" object in the dictionary and the trait variable that is created later on in this function https://github.com/tomdcsmith/eva-cttv-pipeline/blob/clinvar_trait_hints/bin/clinvar_jsons/parse_trait_names.py#L58